bpm/bonita-core/bonita-process-definition/bonita-process-definition-impl/src/main/java/org/bonitasoft/engine/core/process/definition/ProcessDefinitionServiceImpl.java

```diff
@@ -1037,9 +1037,13 @@ public long getNumberOfProcessDeploymentInfosWithAssignedOrPendingHumanTasks(fin
         return persistenceService.searchEntity(SProcessDefinitionDeployInfo.class, "WithAssignedOrPendingHumanTasks", queryOptions, null);
     }
 
-    protected DesignProcessDefinition getDesignProcessDefinition(long processDefinitionId) throws IOException, XMLParseException,
-            SProcessDefinitionNotFoundException, SProcessDefinitionReadException {
-        return processDefinitionBARContribution.convertXmlToProcess(getProcessDeploymentInfo(processDefinitionId).getDesignContent().getContent());
+    @Override
+    public DesignProcessDefinition getDesignProcessDefinition(long processDefinitionId) throws SProcessDefinitionNotFoundException {
```

Comment:
Leave **SProcessDefinitionReadException** pass, add them in method signature.
Convert IOException and XMLParseException into **SProcessDefinitionReadException**